client/src/standalone/videos/embed.ts

```diff
@@ -56,8 +66,60 @@ export class PeerTubeEmbed {
     return window.location.origin + '/api/v1/videos/' + id
   }
 
+  refreshFetch (url: string, options?: Object) {
+    return fetch(url, options)
+      .then((res: Response) => {
+        if (res.status !== 401) return res
+
+        // 401 unauthorized is not catch-ed, but then-ed
+        const error = res
+
+        let refreshingTokenPromise = new Promise((resolve, reject) => {
+          const clientId: string = peertubeLocalStorage.getItem(this.LOCAL_STORAGE_OAUTH_CLIENT_KEYS.CLIENT_ID)
+          const clientSecret: string = peertubeLocalStorage.getItem(this.LOCAL_STORAGE_OAUTH_CLIENT_KEYS.CLIENT_SECRET)
+          const headers = new Headers()
+          headers.set('Content-Type', 'application/x-www-form-urlencoded')
+          const data = {
+            refresh_token: this.user.getRefreshToken(),
+            client_id: clientId,
+            client_secret: clientSecret,
+            response_type: 'code',
+            grant_type: 'refresh_token'
+          }
+
+          fetch('/api/v1/users/token', {
+            headers,
+            method: 'POST',
+            body: objectToUrlEncoded(data)
+          })
+            .then(async (res: Response) => {
+              const obj: UserRefreshToken = await res.json()
```

Comment:
Don't use async/await if you use then

```suggestion
             .then(res => res.json())
             .then((obj: UserRefreshToken) => { ...
```